napari/layers/tracks/tracks.py

```diff
@@ -458,7 +458,9 @@ def tail_length(self) -> Union[int, float]:
 
     @tail_length.setter
     def tail_length(self, tail_length: Union[int, float]):
-        self._tail_length = np.clip(tail_length, 1, self._max_length)
+        if tail_length > self._max_length:
+            self._max_length = tail_length
```

Comment:
here you update class variable. You should not do this as it will impact all class instances. 

To fix this you need to at least move `_max_length` definition to constructor.

Reply:
> here you update class variable.

fwiw, I'm not sure that's true in this case, because he did `self.___`:

```python
In [1]: class T:
   ...:     x = 1
   ...:

In [2]: t1 = T()

In [3]: t2 = T()

In [4]: t1.x
Out[4]: 1

In [5]: t2.x
Out[5]: 1

In [6]: T.x
Out[6]: 1

In [7]: t1.x = 2

In [8]: t1.x
Out[8]: 2

In [9]: t2.x
Out[9]: 1

In [10]: T.x
Out[10]: 1
```

Reply:
I see a single `_` but maybe this is some part of python tricks and may be problematic to many contributors. I'm pretty sure that I see a scenario where it modifies the class variable.  

I still prefer to move this to the constructor.

Reply:
no it's not about the underscore, the point is that attribute access happens _first_ on the object, and then falls back to the class.  So, if you don't use `self.something = 'seomthing'` then it will get the class attribute, but once you set `self.something`... then you're setting it on the instance, not on the class.  
In any case, I agree, it's tricky, and it's just fine to put it in the constructor.  But didn't want anyone to get the wrong idea about how attribute access works :)